python/cudf/dataframe/multiindex.py

```diff
@@ -336,6 +339,53 @@ def __getitem__(self, index):
     def _values(self):
         return list([i for i in self])
 
+    @classmethod
+    def _concat(cls, objs):
+        _need_codes = False
+        from cudf import DataFrame
+        from cudf import MultiIndex
+        for idx, o in enumerate(objs):
+            if not hasattr(o, '_source_data'):
+                _need_codes = True
+        if _need_codes:
```

Comment:
Unless there's a compelling reason, lets just make this raise `NotImplementedError` for now until we can handle this more efficiently or avoid it entirely 😄